consensus/viewchange/roster/mod.go

```diff
@@ -122,6 +142,42 @@ func (r roster) Apply(changeset viewchange.ChangeSet) viewchange.EvolvableAuthor
 	return roster
 }
 
+// Diff implements viewchange.Authority. It returns the change set that must be
+// applied to the current authority to get the given one.
+func (r roster) Diff(o viewchange.Authority) viewchange.ChangeSet {
+	changeset := viewchange.ChangeSet{}
+
+	other, ok := o.(roster)
+	if !ok {
+		return changeset
+	}
+
+	i := 0
+	k := 0
+	for i < len(r.addrs) || k < len(other.addrs) {
+		if i < len(r.addrs) && k < len(other.addrs) {
+			if r.addrs[i].Equal(other.addrs[k]) {
```

Comment:
The order is important for the collective signature for instance, so even if they are unsorted for some reasons, the change set should correct it so that when you follow the links, you'll verify with the right aggregate.

Reply:
ok 👍